wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java

```diff
@@ -260,34 +263,67 @@ public NodeList getFileData(String logicalname, Long beginrow, Integer numrows,
     }
 
     /**
-     * Use this function to retrieve file metadata such as column information
+     * Use this function to retrieve file metadata such as column information,
+     * for superfiles the metadata from the first subfile will be returned.
      * @param logicalname    - Logical filename.
      * @param clustername    - Optional - The cluster the logical filename is associated with.
-     * @return Array of DFUDataColumns
+     * @return ArrayList of DFUDataColumnInfo
      * @throws Exception
      */
-    public DFUDataColumn[] getFileMetaData(String logicalname, String clustername) throws Exception
+    public List<DFUDataColumnInfo> getFileMetaData(String logicalname, String clustername) throws Exception
```

Comment:
the new signature makes sense, but I'm concerned about changing an interface like this, specially in a minor release. Does it make sense to overload the method w/ both signatures 
public List<DFUDataColumnInfo> getFileMetaData(String logicalname, String clustername) throws Exception
and
public DFUDataColumn[] getFileMetaData(String logicalname, String clustername) throws Exception

Reply:
I checked the call heirarchy for this method and currently it's only called by wsclient getFileDetails() and two unit tests in wsclient. It's not used in HIPIE, DSP or any other ln products I'm aware of.

It is a bit draconian to change the interface in a minor release. We can't have the same method signature returning two different objects though, so a safer bet would be to add a  getFileMetadataInfo method returning the new list.  I'll add a deprecated flag to the old method.